parseclj-parser.el

```diff
@@ -203,15 +203,21 @@ functions. Additionally the following options are recognized
        (t (push token stack)))
 
       ;; Reduce based on top two items on the stack (special prefixed elements)
-      (let* ((top-value (parseclj--take-value stack value-p))
-             (opening-token (parseclj--take-token (nthcdr (length top-value) stack) value-p '(:discard :tag)))
-             (new-stack (nthcdr (+ (length top-value) (length opening-token)) stack)))
-        (when (and top-value opening-token)
+      (let (top-value
+            opening-token
+            new-stack)
+        (setq top-value (parseclj--take-value stack value-p))
+        (setq opening-token (parseclj--take-token (nthcdr (length top-value) stack) value-p '(:discard :tag)))
```

Comment:
I'd prefer to leave these in the `let`, that would also make it a bit easier to review. Right now it's hard to see what exactly changed here.

Reply:
You're right, I screwed up while reworking the code.